### DISCUSSION TITLE
Scheduling different tasks within the same application

### DISCUSSION THREAD

**Discussion by SoftCircuits** [dahall/TaskScheduler#972] [Q&A] [answered]
I'm trying to determine if this component will work for our purposes.

It looks like all of our task will likely be within the same executable file. I can easily imagine using command line arguments to specify which task is to be run. But it might be a problem if the application was asked to run two tasks at the same time.

Does anyone have experience doing this? Does this make sense or am I missing a better way? There wouldn't be any way to prevent the app from being asked to perform multiple tasks at the same time is there?

**Comment by dahall** [comment]
This library has no problem running parallel tasks. However, it does establish all tasks as tasks within the OS, all persistent outside of app lifetimes, visible to all users and applications, and removeable by administrators. If you just need tasks to run while your app is live, you may want to look at some other methods.

**Reply by SoftCircuits** [reply, marked as answer]
Thanks for responding.

To clarify, I want my tasks to run no matter what. I would be calling your library from a website. But the tasks themselves are within a console application.

If I understood you correctly, your answer is simply that the Windows Task Manager doesn't have any problems running multiple instances of my console application. Obviously, I guess it would be up to me to ensure the console application doesn't have problems running multiple instances at the same time.
